SU2_CFD/src/solvers/CTurbSolver.cpp

```diff
@@ -674,21 +673,15 @@ void CTurbSolver::ComputeUnderRelaxationFactor(CSolver **solver_container, const
         /* We impose a limit on the maximum percentage that the
          turbulence variables can change over a nonlinear iteration. */
 
-        const unsigned long index = iPoint*nVar + iVar;
-        su2double ratio = LinSysSol[index]/(nodes->GetSolution(iPoint, iVar)+EPS);
-        if (ratio > allowableIncrease) {
-          localUnderRelaxation = min(allowableIncrease/ratio, localUnderRelaxation);
-        } else if (ratio < allowableDecrease) {
-          localUnderRelaxation = min(fabs(allowableDecrease)/ratio, localUnderRelaxation);
+        const unsigned long index = iPoint * nVar + iVar;
+        su2double ratio = fabs(LinSysSol[index]) / (nodes->GetSolution(iPoint, iVar) + EPS);
+        if (ratio > allowableRatio) {
+          localUnderRelaxation = min(allowableRatio / ratio, localUnderRelaxation);
         }
 
       }
     }
 
-    /* Choose the minimum factor between mean flow and turbulence. */
-
-    localUnderRelaxation = min(localUnderRelaxation, solver_container[FLOW_SOL]->GetNodes()->GetUnderRelaxation(iPoint));
-
```

Comment:
Does this, and the change in CFVMFlowSolverBase mean that there are two different underrelaxation factors, one for the mean solver, and one for the turbulent solver?

Reply:
One of them needs to go, otherwise l guess we have the cyclic dependency problem @bmunguia mentioned.